mbed_lstools/lstools_darwin.py

```diff
@@ -26,134 +26,86 @@
 
 logger = logging.getLogger("mbedls.lstools_darwin")
 
+def _find_TTY(obj):
+    ''' Find the first tty (AKA IODialinDevice) that we can find in the
+        children of the specified object, or None if no tty is present.
+    '''
+    if 'IODialinDevice' in obj:
+        return obj['IODialinDevice']
+    if 'IORegistryEntryChildren' in obj:
+        for child in obj['IORegistryEntryChildren']:
+            found = _find_TTY(child)
+            if found:
+                return found
+    return None
+
 class MbedLsToolsDarwin(MbedLsToolsBase):
-    """ MbedLsToolsDarwin supports mbed-enabled platforms detection on Mac OS X
+    """ mbed-enabled platform detection on Mac OS X
     """
 
-    mbed_volume_name_match = re.compile(r'(\bmbed\b|\bSEGGER MSD\b)', re.I)
-
-    def list_mbeds(self):
-        """ returns mbed list with platform names if possible
-        """
-
-        result = []
+    def __init__(self, **kwargs):
+        MbedLsToolsBase.__init__(self, **kwargs)
+        self.mbed_volume_name_match = re.compile(r'(\bmbed\b|\bSEGGER MSD\b)',
+                                                 re.I)
+        self.mac_version = float('.'.join(platform.mac_ver()[0].split('.')[:2]))
 
+    def find_candidates(self):
         # {volume_id: {serial:, vendor_id:, product_id:, tty:}}
-        volumes = self.get_mbed_volumes()
+        volumes = self.get_volumes()
 
         # {volume_id: mount_point}
         mounts = self.get_mount_points()
-
-        volumes_keys = set(volumes.keys())
-        mounts_keys = set(mounts.keys())
-        intersection = volumes_keys & mounts_keys
-
-        valid_volumes = {}
-
-        for key in intersection:
-            valid_volumes[key] = volumes[key]
-
-        # put together all of that info into the expected format:
-        result =  [
+        return [
             {
                 'mount_point': mounts[v],
                 'serial_port': volumes[v]['tty'],
-                  'target_id': self.target_id(volumes[v]),
-              'platform_name': self.platform_name(self.target_id(volumes[v]))
-            } for v in valid_volumes
+                  'target_id': volumes[v].get('serial')
```

Comment:
Looks like indentation got a little messed up here

Reply:
I'll change the others, they were wrong before.